sihl/http/request.ml

```diff
@@ -0,0 +1,374 @@
+(* This module is based on the reworked API of Opium.
+
+   MIT License
+
+   Copyright (c) 2019 Rudi Grinberg
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy of this
+   software and associated documentation files (the "Software"), to deal in the Software
+   without restriction, including without limitation the rights to use, copy, modify,
+   merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+   permit persons to whom the Software is furnished to do so, subject to the following
+   conditions:
+
+   The above copyright notice and this permission notice shall be included in all copies
+   or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+   INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+   PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+   HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+   CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR
+   THE USE OR OTHER DEALINGS IN THE SOFTWARE. *)
+
+open Lwt.Syntax
+include Opium_kernel.Rock.Request
+
+(* TODO [jerben] expose as core middleware *)
+let id_key : string Opium_kernel.Hmap.key =
+  Opium_kernel.Hmap.Key.create ("id", Sexplib.Std.sexp_of_string)
+;;
+
+let make
```

Comment:
Allow this to take `~meth` so any type of request can be given.

Reply:
I removed `make` since it is tricky to allow polymorphic variants without enumerating all.

Reply:
I now still added `meth`.